nats-connector/index.ls

```diff
@@ -0,0 +1,36 @@
+require! {
+  'nats': NATS
+  'prelude-ls': {each, obj-to-pairs}
+  'util'
+}
+
+class NatsConnector
+
+  ->
+    @nats = NATS.connect "nats://#{process.env.NATS_HOST}:4222"
+    console.log "nats: connected"
+
+
+  subscribeMapping: (messageCallbackMapping) ->
+    messageCallbackMapping |> obj-to-pairs |> each ([name, callback]) ~>
+      @nats.subscribe name, (requestDataStr, replyTo) ~>
+        requestData = JSON.parse requestDataStr
+        @_log 'received request for', name, requestData
+        callback requestData, reply: (replyData) ~>
+          @_log 'responding to', name, replyData
+          @nats.publish replyTo, JSON.stringify(replyData)
```

Comment:
If you want the reply to also broadcast, (similar to what exocom does) this would need two publish calls.